compiler/semgnrc.nim

```diff
@@ -356,13 +360,25 @@ proc semGenericStmt(c: PContext, n: PNode,
   of nkCaseStmt:
     openScope(c)
     n[0] = semGenericStmt(c, n[0], flags, ctx)
+    var errors: CandidateErrors
+    let r = resolveCaseSmtMacroOverload(c, n, errors)
```

Comment:
Too early. Instead you need to special case `of` branches to allow for unknown identifiers.

Reply:
There are two parts of this:
1. To allow the syntax. My attempt tries to see if the `case(exp)` matches the macro. 
2. The macro needs to be expanded so you account for the `@` identifiers (`a`, `b`) below
```nim
  proc regularCase[T]() = 
    let test = () => int is typeof(T)
    case test():
    let foo = [(1, 3), (3, 4)]
    case foo:
    of [(1, @a), (_, @b)]:
      echo a, b
    else: discard
    
```
Your comment is in reference to [1] or to both?